apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Logging/LabellingScope.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright 2021 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Logging.V2;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// Represents the data for the labelling scope, basically, labels.
+    /// </summary>
+    // Note: We implement IEnumerable for the benefit of other ILogger implementations,
+    // so they can represent this information as they see fit.
+    public class LabellingScopeState : IEnumerable<KeyValuePair<string, object>>
+    {
+        /// <summary>
+        /// Creates a new <see cref="LabellingScopeState"/> for the given labels.
+        /// </summary>
+        public LabellingScopeState(IEnumerable<KeyValuePair<string, string>> labels)
+        {
+            var dictionary = new Dictionary<string, string>();
+            if (labels != null)
+            {
+                foreach(var pair in labels)
+                {
+                    // Let's do this to allow overriding
+                    dictionary[pair.Key] = pair.Value;
+                }
+            }
+            Labels = new ReadOnlyDictionary<string, string>(dictionary);
+        }
+
+        internal LabellingScopeState(params KeyValuePair<string, string>[] labels)
+            :this((IEnumerable<KeyValuePair<string, string>>)labels)
```

Comment:
Nit: space after colon. Might be simpler as `: this(labels.AsEnumerable())` too...

Reply:
Done, both.